tests/ui/from_str_radix_10.stderr

```diff
@@ -0,0 +1,46 @@
+error: this call to `from_str_radix` can be replaced with a call to `str::parse`
+  --> $DIR/from_str_radix_10.rs:28:5
+   |
+LL |     u32::from_str_radix("30", 10)?;
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `"30".parse::<u32>()`
+   |
+   = note: `-D clippy::from-str-radix-10` implied by `-D warnings`
+
+error: this call to `from_str_radix` can be replaced with a call to `str::parse`
+  --> $DIR/from_str_radix_10.rs:29:5
+   |
+LL |     i64::from_str_radix("24", 10)?;
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `"24".parse::<i64>()`
+
+error: this call to `from_str_radix` can be replaced with a call to `str::parse`
+  --> $DIR/from_str_radix_10.rs:30:5
+   |
+LL |     isize::from_str_radix("100", 10)?;
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `"100".parse::<isize>()`
+
+error: this call to `from_str_radix` can be replaced with a call to `str::parse`
+  --> $DIR/from_str_radix_10.rs:31:5
+   |
+LL |     u8::from_str_radix("7", 10)?;
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `"7".parse::<u8>()`
+
+error: this call to `from_str_radix` can be replaced with a call to `str::parse`
+  --> $DIR/from_str_radix_10.rs:32:5
+   |
+LL |     u16::from_str_radix(&("10".to_owned() + "5"), 10)?;
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `(&("10".to_owned() + "5")).parse::<u16>()`
```

Comment:
Does this suggestion make sense? Clippy doesn't currently lint against it, and I think it might not always be correct to suggest using `("10".to_owned() + "5").parse::<u16>` instead, so perhaps it should be left as is along with the current `Applicability::MaybeIncorrect`?

Reply:
Why would it be incorrect?

Reply:
Not incorrect per se - just that I'm not sure if the applicability level is `MachineApplicable` because in this case the suggestion can be shortened to `("10".to_owned() + "5").parse::<u16>`, which might be a bit harder to catch. Maybe it's too much of a nit 😅

Reply:
Honestly now that I've thought about it more the same could be said about the turbofish, so I don't see this being a problem.

Reply:
Ah, it's about the reference. Yes, that may make a difference, and we could alleviate that by checking the method receiver and replace a possible `&` expression by its operand.

Reply:
What do you mean by checking the method receiver?

Reply:
That would be `arguments[0]` in the code.

Reply:
Oh. Then I suppose that adding a check to see if the type implements `Deref<Target = str>` is sufficient, I'll do that

Reply:
@llogiq - I couldn't find a way to check if a type implements `Deref<Target = str>` specifically; Checking for a `Deref` implementation is easy, but I'm having a hard time accessing the associated type. Would it be okay if I implemented this enhancement for `String` only, as it probably represents 99% of the cases where the lint will show up anyway? The suggestion is correct in any case, so it shouldn't matter too much.

Reply:
Yes, that would be OK. We should note this shortcoming under the "Problems" section in the docs, though. Best to be candid with that stuff.